### PR TITLE
Migrate from sql/parser to sql/parser2.

### PR DESCRIPTION
**Pull request by petermattis**
Fixed parser error generation for SQL statements that span multiple
lines.

Removed unused sqlwire.Datum functionality. The parser and expression
evaluator no longer use unsigned integers. And the expression evaluator
no longer uses sqlwire.Datum directly.

A follow-on change will rename sql/parser2 to sql/parser.

### CODE REVIEW

File: sql/sqlserver/server.go [hunk 1]
```diff
@@ -256,13 +258,14 @@ func (s *Server) ShowIndex(session *Session, p *parser.ShowIndex, args []sqlwire
 	// TODO(pmattis): This output doesn't match up with MySQL. Should it?
 	var rows []sqlwire.Result_Row
 
+	name := p.Table[len(p.Table)-1]
```
Comment (tamird) [inline sql/sqlserver/server.go]: `.Table()`?

**Comment by tamird** [comment]
LGTM

File: sql/parser2/expr.go [hunk 2]
```diff
@@ -234,6 +234,19 @@ func (NullVal) String() string {
...
+func (n QualifiedName) Database() string {
+	if len(n) > 1 {
+		return n[0]
+	}
+	return ""
+}
+
+// Table returns the table portion of the name.
+func (n QualifiedName) Table() string {
+	return n[len(n)-1]
```
Comment (tschottdorf) [inline sql/parser2/expr.go]: only one char?

File: sql/sqlserver/server.go [hunk 3]
```diff
@@ -563,25 +561,51 @@ func (s *Server) Select(session *Session, p *parser.Select, args []sqlwire.Datum
 	return nil
 }
 
+// Set sets session variables.
+func (s *Server) Set(session *Session, p *parser2.Set, args []sqlwire.Datum,
+	resp *sqlwire.Response) error {
+	// By using QualifiedName.String() here any variables that are keywords will
+	// be double quoted.
```
Comment (tschottdorf) [inline sql/sqlserver/server.go]: doesn't SQL generally use single-quotes?

File: sql/sqlserver/server.go (hunk 1 above)
Comment (petermattis) [inline sql/sqlserver/server.go]: Definitely. I originally started without the `QualifiedName.{Database,Table}()` accessors, but the code was really confusing. Missed this one place.

File: sql/sqlserver/server.go (hunk 3 above)
Comment (petermattis) [inline sql/sqlserver/server.go]: `QualifiedName` is a dot separated list of identifiers. Double quotes are used for identifiers that are also keywords (such as "database"). Single quotes are used for strings.

File: sql/parser2/expr.go (hunk 2 above)
Comment (petermattis) [inline sql/parser2/expr.go]: Oops, forgot to reply to this. `QualifiedName` is `[]string`. This is selecting the last element of the slice.

**Pull request merged**